camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/mapred/EtlRecordReaderForUnitTest.java

```diff
@@ -0,0 +1,59 @@
+package com.linkedin.camus.etl.kafka.mapred;
+
+import java.io.IOException;
+
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.easymock.EasyMock;
+
+import com.linkedin.camus.coders.CamusWrapper;
+import com.linkedin.camus.coders.MessageDecoder;
+import com.linkedin.camus.etl.kafka.coders.MessageDecoderFactory;
+import com.linkedin.camus.schemaregistry.SchemaNotFoundException;
+
+
+public class EtlRecordReaderForUnitTest extends EtlRecordReader {
+
+  public static enum DecoderType {
+    REGULAR,
```

Comment:
same idea as EtlInputFormatForUnitTest. use enum to indicate desired behavior (in this case, what decoder should this class use).